extensions/azurecore/src/azureResource/tree/treeProvider.ts

```diff
@@ -66,6 +75,23 @@ export class AzureResourceTreeProvider implements TreeDataProvider<TreeNode>, IA
 		}
 	}
 
+	private async loadAccounts(): Promise<void> {
+		try {
+			this.accounts = await this.appContext.getService<IAzureResourceAccountService>(AzureResourceServiceNames.accountService).getAccounts();
+			// System has been initialized
+			this.setSystemInitialized();
+			this._onDidChangeTreeData.fire(undefined);
+		} catch (err) {
+			// Skip for now, we can assume that the accounts changed event will eventually notify instead
+			this.isSystemInitialized = false;
+		}
+	}
+
+	private setSystemInitialized() {
+		this.isSystemInitialized = true;
+		this.loadingAccountsPromise = undefined;
+	}
```

Comment:
Without having some sort of cancel isn't the promise still going to run and eventually return even if the onDidChangeAccounts is called first? What if in loadAccounts you checked after getAccounts finishes and see if isSystemInitialized is false and only set this.accounts to the returned accounts if that's the case?

Reply:
So if the call succeeds, the account manager is up & this is treated as loaded. This is still equivalent to the current behavior - on promise completion it'll cause getChildren to be called again, the accounts will be queried freshly & things "should" work. I'm going to won't fix this comment if that's OK.